framework/src/controller/controller.js

```diff
@@ -160,54 +155,62 @@ module.exports = class Controller {
 		);
 	}
 
-	async _loadModules() {
-		return Promise.each(Object.keys(this.modules), m =>
-			this._loadInMemoryModule(m)
+	async _loadModules(modules) {
+		return Promise.all(
+			Object.keys(modules).map(alias =>
+				this._loadInMemoryModule(
+					alias,
+					modules[alias].klass,
+					modules[alias].options
+				)
+			)
 		);
 	}
 
-	async _loadInMemoryModule(alias) {
-		const module = this.modules[alias].spec;
-		const moduleConfig = this.modules[alias].config;
+	async _loadInMemoryModule(alias, Klass, options) {
+		const module = new Klass(options);
 		validateModuleSpec(module);
 
+		const moduleAlias = alias || module.constructor.alias;
+		const { name, version } = module.constructor.info;
+
 		this.logger.info(
-			`Loading module with alias: ${module.alias}(${module.info.name}:${
-				module.info.version
-			})`
+			`Loading module with alias: ${moduleAlias}(${name}:${version})`
 		);
 
 		const channel = new EventEmitterChannel(
-			module.alias,
+			moduleAlias,
 			module.events,
 			Object.keys(module.actions),
 			this.bus,
 			{}
 		);
-		this.channels[alias] = channel;
+		this.modulesChannels[moduleAlias] = channel;
 
 		await channel.registerToBus();
 
 		Object.keys(module.actions).forEach(action => {
 			channel.action(action, module.actions[action]);
 		});
 
-		channel.publish(`${module.alias}:registeredToBus`);
-		channel.publish(`${module.alias}:loading:started`);
-		await module.load(channel, moduleConfig);
-		channel.publish(`${module.alias}:loading:finished`);
+		channel.publish(`${moduleAlias}:registeredToBus`);
+		channel.publish(`${moduleAlias}:loading:started`);
+		await module.load(channel);
+		channel.publish(`${moduleAlias}:loading:finished`);
+
+		this.modules[moduleAlias] = module;
 		this.logger.info(
-			`Module ready with alias: ${module.alias}(${module.info.name}:${
-				module.info.version
-			})`
+			`Module ready with alias: ${moduleAlias}(${name}:${version})`
 		);
 	}
 
 	async unloadModules(modules = null) {
-		return Promise.mapSeries(modules || Object.keys(this.modules), async m => {
-			await this.modules[m].spec.unload();
-			delete this.modules[m];
-		});
+		return Promise.all(
+			(modules || Object.keys(this.modules)).map(async alias => {
```

Comment:
I don't think this is behaving as you expect.
Do you want to run the `unload`'s in parallel?